terraria-1.4-tmodloader/terraria-1.4-tmodloader.json

```diff
@@ -0,0 +1,229 @@
+{
+  "name": "terraria-1.4-tmodloader",
+  "display": "Terraria 1.4 - tModLoader",
+  "type": "terraria",
+  "install": [
+    {
+      "files": [
+        "https://github.com/tModLoader/tModLoader/releases/download/${version}/tModLoader.zip"
+      ],
+      "type": "download"
+    },
+    {
+      "target": "EnableAllMods.sh",
+      "text": "#!/bin/bash\nif [ \"$1\" == \"false\" ]; then\n  exit 0\nfi\n\nenabledMods=\"[\"\n\nfor file in ./mods/*.tmod; do\n  modName=\"$(basename \"$file\" .tmod)\"\n  enabledMods=\"$enabledMods\\n\\t\\\"$modName\\\",\"\n  echo \"$modName\"\ndone\n\nenabledMods=\"${enabledMods::-1}\\n]\"\necho -e \"Enabled Mods: $enabledMods\"\necho -e \"$enabledMods\" > \"mods/enabled.json\"",
+      "type": "writefile"
+    },
+    {
+      "target": "serverconfig.txt",
+      "text": "world=../${wp}/${world}.wld\nautocreate=${size}\nseed=${seed}\nworldname=${world}\ndifficulty=${difficulty}\nmaxplayers=${players}\nport=${port}\npassword=${password}\nmotd=${motd}\nworldpath=../${wp}\nbanlist=../${ban}\nsecure=${secure}\nlanguage=${lang}\nupnp=${upnp}\nnpcstream=${npcs}\npriority=${priority}\nslowliquids=${sl}\nworldrollbackstokeep=${backups}\nmodpath=../${mp}",
+      "type": "writefile"
+    },
+    {
+      "target": "enabled.json",
+      "text": "[\n    \"*\"\n]",
+      "type": "writefile"
+    },
+    {
+      "commands": [
+        "rm -r ./server",
+        "mkdir ./server",
+        "unzip tModLoader.zip -d ./server",
+        "chmod +x ./server",
+        "chmod +x ./server/LaunchUtils/ScriptCaller.sh",
+        "mkdir ./${wp}",
+        "mkdir ./${mp}",
+        "rm tModLoader.zip",
+        "mv ./enabled.json ./${mp}/enabled.json",
+        "chmod +x ./EnableAllMods.sh"
+      ],
+      "type": "command"
+    }
+  ],
+  "run": {
+    "stop": "exit",
+    "command": "server/LaunchUtils/ScriptCaller.sh -server -ip ${ip} ${disableannouncementbox} -announcementboxrange ${abr} -config ../serverconfig.txt",
+    "workingDirectory": "",
+    "pre": [
+      {
+        "commands": [
+          "./EnableAllMods.sh ${autoEnableMods} "
+        ],
+        "type": "command"
+      }
+    ],
+    "post": [],
+    "environmentVars": {}
+  },
+  "data": {
+    "abr": {
+      "type": "string",
+      "desc": "Sets the announcementbox text messaging range in pixels, -1 for serverwide announcements.",
+      "display": "announcementbox-range",
+      "required": true,
+      "value": "-1"
```

Comment:
Make any number fields a "type": "integer"

Reply:
Changed any number value to integer in https://github.com/PufferPanel/templates/pull/164/commits/04d2fac6a53d24b06a83d43db7242634708357ae